### ISSUE TITLE
Performance of scope.$watch

### ISSUE THREAD

**Issue by dunksmith** [beezee/angular-dotdotot#2]
Using scope.$watch without passing an expression very inefficent, since it fires on _any_ scope change (whether relevent to the plugin or not).  Profiling in Chrome showed that this was causing dotdotdot to eat up CPU time.

Maybe you could at least mention this in the readme to avoid people getting caught out?

Here's my workaround - not great as the scope variable is hardcoded, but hopefully you get the general idea.

```
app.directive('dotdotdot', ['$timeout', function($timeout){
    return {
        restrict: 'A',
        scope: true,    // Pass scope in
        link: function(scope, element, attributes) {

            // Hardcode watch expression (must be a better way??)
            scope.$watch('scopeVarThatAffectsSize', function() {

                // Wait for element to render - 400 appears to be a good number from testing
                $timeout(function() {
                    // Don't use { watch: true } here as it polls continuously
                    element.dotdotdot();
                }, 400);
            });
        }
    }
}]);
```

**Comment by ecollis6** [comment]
I completely agree with this. I just implemented this in a current project with the following modifications:

<b>Passed my ng-model (from my input) into "content" attribute:</b>

<pre>
scope: {
    content: '='
}
</pre>

<b>Set watch on "content":</b>

<pre>
scope.$watch('content', function() {
    ...
});
</pre>

<b>Set debounce effect on my input to increase performance:</b>

<pre>
&lt;input ng-model=&quot;vm.InputValue&quot; ng-model-options=&quot;{debounce: {&#39;default&#39;: 100}}&quot;/&gt;
</pre>

Here is the directive being used:

<pre>
&lt;div dotdotdot content=&quot;vm.InputValue&quot;/&gt;
</pre>
